spyder/plugins/projects/plugin.py

```diff
@@ -257,6 +272,15 @@ def on_main_menu_teardown(self):
         main_menu = self.get_plugin(Plugins.MainMenu)
         main_menu.remove_application_menu(ApplicationMenus.Projects)
 
+    @on_plugin_teardown(plugin=Plugins.Switcher)
+    def on_switcher_teardown(self):
+        # Disconnect from switcher
+        self._switcher.sig_mode_selected.disconnect(self.handle_switcher_modes)
+        self._switcher.sig_item_selected.disconnect(
+            self.handle_switcher_selection)
+        self._switcher.sig_search_text_available.disconnect(
+            self.handle_switcher_results)
```

Comment:
```suggestion
        self._switcher.sig_mode_selected.disconnect(self._handle_switcher_modes)
        self._switcher.sig_item_selected.disconnect(
            self._handle_switcher_selection)
        self._switcher.sig_search_text_available.disconnect(
            self._handle_switcher_results)
        self._switcher = None
```

* This follows the renaming of methods I suggested above.
* It also remove the reference to the Switcher plugin.